src/eval.rs

```diff
@@ -289,6 +309,139 @@ impl<'tcx> MainThreadState<'tcx> {
         }
         interp_ok(Poll::Pending)
     }
+
+    fn eval_entry(&mut self, ecx: &mut MiriInterpCx<'tcx>) -> InterpResult<'tcx, ()> {
```

Comment:
Please move all this code into a function on `MiriInterpCx`, right after `create_ecx`, so that the diff remains small.

Also, please call the function `call_main`. The function does not do the actual evaluation so `eval_entry` does not make a lot of sense as a name.